server/context.ts

```diff
@@ -74,40 +78,20 @@ export async function createContext({ req }: { req: Request }) {
   }
 }
 
-async function getAndUpdateSession(
+function getSessionDataFromToken(
   token: string,
   updates: Pick<Session, "userAgent" | "deviceNonce" | "ip" | "countryCode">
-): Promise<Session> {
+): Session {
   const { sub: userId, session_id: sessionId, sessionData } = decodeJwt(token);
 
-  const sessionUpdates: Partial<typeof updates> = {};
-  for (const [key, value] of Object.entries(updates) as [
-    keyof typeof updates,
-    string
-  ][]) {
-    if (value && sessionData?.[key] !== value) {
-      sessionUpdates[key] = value;
-    }
-  }
-
-  if (Object.keys(sessionUpdates).length > 0) {
-    console.log("Updating session:", sessionUpdates);
-
-    prisma.session
-      .update({
-        where: { id: sessionId },
-        data: sessionUpdates,
-      })
-      .catch((error) => {
-        console.error("Error updating session:", error);
-      });
-  }
+  // We don't need to update the session in the database
+  // The database triggers will handle that automatically when Supabase updates the session
```

Comment:
@matteyu If this is removed then `ip` and `userAgent` updates will only happen when supabase session is refreshed. Also, we are updating the `deviceNonce` value to session so it is not updated as well with this change. I think lets revert to the previous change and we can revisit this just after this is merged.

Reply:
@matteyu As Pawan mentioned, that code needs to be kept for now, even though it looks redundant. I've merged 2 other PRs related to WorkKeyShare's foreign keys and challenges that were also related to this, so there are some conflicts on this PR now. Once you resolve this issue and the conflicts, we'll merge it, and after that we'll see how to simplify these issues we are having with keeping session data updated.